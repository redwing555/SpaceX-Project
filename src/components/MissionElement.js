import { useSelector, useDispatch } from 'react-redux';
import { useEffect } from 'react';
import { fetchAllMissions } from '../redux/missions/missions';
import IndividualMission from './IndividualMission';

const MissionElement = () => {
  const dispatch = useDispatch();
  const missionsData = useSelector((state) => state.missionsReducer);
  useEffect(() => {
    dispatch(fetchAllMissions());
  }, []);
  return (
    <>
      {missionsData.map((item) => (<IndividualMission key={item.mission_id} data={item} />
      ))}
    </>
  );
};
export default MissionElement;
