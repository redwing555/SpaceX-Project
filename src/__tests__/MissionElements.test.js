import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import MissionElement from '../components/MissionElement';

describe('Mission Element Tests', () => {
  test('Check if Component Renders in DOM', () => {
    render(
      <Provider store={store}>
        <MissionElement />
      </Provider>,
    );
    expect(screen.queryByText(/Rocket/)).toBeNull();
    expect(screen.queryByText(/Mission/)).toBeTruthy();
  });
  // it('renders correctly', () => {
  //   const tree = renderer
  //     .create(
  //       <Router>
  //         <NavigationLinks />
  //       </Router>,
  //     )
  //     .toJSON();
  //   expect(tree).toMatchSnapshot();
  // });
});
