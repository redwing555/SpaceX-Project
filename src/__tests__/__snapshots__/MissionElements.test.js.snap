// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission Element Tests renders correctly 1`] = `
<div
  className="missions-container"
>
  <span>
    <b>
      Mission
    </b>
  </span>
  <span>
    <b>
      Description 
    </b>
  </span>
  <span>
    <b>
      Status
    </b>
  </span>
  <span>
    
  </span>
</div>
`;
